### PR TITLE
Fix compilation on GCC4

### PR DESCRIPTION
**Pull request by czeslawmakarski**
Backport of master branch fix.

### CODE REVIEW

File: src/nrf_802154_trx.c [hunk 1]
```diff
@@ -177,22 +177,26 @@ static const nrf_802154_fal_event_t m_deactivate_on_disable =
...
 {
-    .type                             = NRF_802154_FAL_EVENT_TYPE_TIMER,
-    .override_ppi                     = false,
-    .event.timer.p_timer_instance     = NRF_802154_TIMER_INSTANCE,
-    .event.timer.compare_channel_mask =
-        ((1 << NRF_TIMER_CC_CHANNEL0) | (1 << NRF_TIMER_CC_CHANNEL2)),
-    .event.timer.counter_value = RX_RAMP_UP_TIME
+    .type         = NRF_802154_FAL_EVENT_TYPE_TIMER,
+    .override_ppi = false,
+    .event_timer  =
```
Comment (ankuns) [inline src/nrf_802154_trx.c]: Is there `event_timer` field? Should it be `event.timer`

Comment (czeslawmakarski) [reply src/nrf_802154_trx.c]: Oops. Fixed.

File: src/nrf_802154_trx.c [hunk 2]
```diff
@@ -177,22 +177,26 @@ static const nrf_802154_fal_event_t m_deactivate_on_disable =
...
-        ((1 << NRF_TIMER_CC_CHANNEL0) | (1 << NRF_TIMER_CC_CHANNEL2)),
-    .event.timer.counter_value = RX_RAMP_UP_TIME
+    .type         = NRF_802154_FAL_EVENT_TYPE_TIMER,
+    .override_ppi = false,
+    .event.timer  =
+    {
+        .p_timer_instance     = NRF_802154_TIMER_INSTANCE,
+        .compare_channel_mask = ((1 << NRF_TIMER_CC_CHANNEL0) | (1 << NRF_TIMER_CC_CHANNEL2)),
+        .counter_value        = RX_RAMP_UP_TIME,
+    },
```
Comment (ankuns) [inline src/nrf_802154_trx.c]: unnecessary comma

File: src/nrf_802154_trx.c [hunk 3]
```diff
@@ -177,22 +177,26 @@ static const nrf_802154_fal_event_t m_deactivate_on_disable =
...
-        ((1 << NRF_TIMER_CC_CHANNEL0) | (1 << NRF_TIMER_CC_CHANNEL2)),
-    .event.timer.counter_value = TX_RAMP_UP_TIME
+    .type         = NRF_802154_FAL_EVENT_TYPE_TIMER,
+    .override_ppi = false,
+    .event.timer  =
+    {
+        .p_timer_instance     = NRF_802154_TIMER_INSTANCE,
+        .compare_channel_mask = ((1 << NRF_TIMER_CC_CHANNEL0) | (1 << NRF_TIMER_CC_CHANNEL2)),
+        .counter_value        = TX_RAMP_UP_TIME,
+    },
```
Comment (ankuns) [inline src/nrf_802154_trx.c]: unnecessary comma

**Pull request merged**